.github/workflows/main.yml

```diff
@@ -416,6 +421,20 @@ jobs:
           RISC0_USE_DOCKER: "1"
       - run: sccache --show-stats
 
+  stable-ceremony:
+    # Tests that the circom files are unchanged from the versions we ran a trusted setup ceremony on
+    if: needs.changes.outputs.stable-ceremony == 'true'
+    needs: changes
+    runs-on: [self-hosted, prod, Linux, cpu]
+    steps:
+      # This is a workaround from: https://github.com/actions/checkout/issues/590#issuecomment-970586842
+      - run: "git checkout -f $(git -c user.name=x -c user.email=x@x commit-tree $(git hash-object -t tree /dev/null) < /dev/null) || :"
+      - uses: actions/checkout@v4
+      - name: Compare circom hashes to golden values
+        run: bash ./circom_stable.sh
+        working-directory: ./.github/scripts
```

Comment:
I recommend we move this to the `check` job, which always runs (doesn't depend on the `changes` job)